go-controller/pkg/nbdb/logical_switch_port.go

```diff
@@ -28,6 +28,32 @@ type LogicalSwitchPort struct {
 	Up               *bool             `ovsdb:"up"`
 }
 
+// LSPCustomField represents custom field names of a LogicalSwitchPort.
+type LSPCustomField string
+
+// LSPCustomFieldEnabled represents the "Enabled" field of a LogicalSwitchPort.
+const LSPCustomFieldEnabled LSPCustomField = "Enabled"
+
+// LSPOption is a function type that modifies a LogicalSwitchPort and returns the name of the field that was modified.
+type LSPOption func(lsp *LogicalSwitchPort) LSPCustomField
+
+// WithLogicalSwitchPortEnabled returns an LSPOption that sets the Enabled field of a LogicalSwitchPort.
+func WithLogicalSwitchPortEnabled(enabled bool) LSPOption {
+	return func(lsp *LogicalSwitchPort) LSPCustomField {
+		lsp.Enabled = &enabled
+		return LSPCustomFieldEnabled
+	}
+}
+
+// ApplyLogicalSwitchPortOpts applies a list of LSPOption functions to a LogicalSwitchPort.
+func ApplyLogicalSwitchPortOpts(lsp *LogicalSwitchPort, lspOpts ...LSPOption) []LSPCustomField {
+	var customFieldsNames []LSPCustomField
+	for _, opt := range lspOpts {
+		customFieldsNames = append(customFieldsNames, opt(lsp))
+	}
+	return customFieldsNames
+}
+
```

Comment:
We can't do this. This is a generated file. It says on the top:
```
// Code generated by "libovsdb.modelgen"
// DO NOT EDIT.
```
Unless you want to implement this on the model generator, we are potentially over-engineering this.

Reply:
ahh I see.. thank you for that, missed that entirely.
Moved it to libovsdb/ops library where it is actually being used.